perllib/FixMyStreet/App/Controller/Photo.pm

```diff
@@ -69,6 +70,23 @@ sub index :LocalRegex('^(c/)?([1-9]\d*)(?:\.(\d+))?(?:\.(full|tn|fp))?\.(?:jpeg|
 
     $c->detach( 'no_photo' ) unless $c->cobrand->allow_photo_display($item, $photo_number); # Should only be for reports, not updates
 
+    my $body_ids;
+    if ( $is_update ) {
+        $c->stash->{non_public} = $item->problem->non_public;
+        $body_ids = $item->problem->bodies_str_ids;
+    } else {
+        $c->stash->{non_public} = $item->non_public;
+        $body_ids = $item->bodies_str_ids;
+    }
+
+    if ($c->stash->{non_public}) {
+        # Check permission
```

Comment:
I think this prevents a user viewing photos on their own `non_public` reports too, which might not be desired behaviour?